### PR TITLE
Get the build and episode lookup working again

### PR DESCRIPTION
**Pull request by vipuldelwadia**
The program could not build on a current JDK, and could not rename anything even when it did build.

## Build

The Gradle build used the `compile`/`testCompile` configurations removed in Gradle 7, and pinned SWT to an x86_64 macOS binary that cannot load on Apple silicon. The build now selects the SWT artifact matching the host OS and architecture, targets JDK 21 via a toolchain, and ships a Gradle wrapper.

Removed the Ant, Ivy, CircleCI, Travis and Docker files. None could run on a current JDK: the Ant packaging targets need `ant-javafx.jar`, dropped from the JDK after Java 10, and `circle.yml` is the CircleCI 1.0 format retired in 2018. Travis only ever built Linux and its `master` build was red, so it is replaced by GitHub Actions rather than kept alongside.

**This loses native bundle packaging** (`.app`, `.exe`, Linux zip), which only `build.xml` produced. `jpackage` is the replacement and is not set up yet.

## Episode lookup

TheTVDB's version 1 API is retired. Its search endpoint answers every query with an empty document, so no show ever resolved. Replaced with TVmaze, which serves the same data as JSON and needs no API key.

TVmaze publishes only the aired ordering, so the DVD ordering preference now always falls back to the aired placement. It was never exposed in the UI.

## Tests and CI

53 tests, all passing, none skipped. Previously 119 with 4 skipped, of which about 60 more never ran behind `RUN_EXTRA_TESTS = false`; those encoded TheTVDB identifiers and orderings, so they were removed with it.

Added `EndToEndRenameTest`, which takes an ugly filename on disk through parsing, provider lookup, listings and the file move, then asserts the renamed file exists. Every existing test passed while the program could not rename a file, so this covers that gap.

The new Actions workflow builds and tests on Linux, macOS and Windows. SWT loads a different native library on each, so one green platform says little about the others. Each job also launches the GUI and screenshots it, because no test touches SWT: a passing build does not show that a window can open, and a missing native library surfaces only on launch.

## Verified

| Platform | SWT artifact | `gradlew build` | GUI launches |
| --- | --- | --- | --- |
| macOS 15, Apple silicon (local) | `cocoa.macosx.aarch64` | pass | yes, menus and a real rename both work |
| macOS x86_64 (CI) | `cocoa.macosx.x86_64` | pass | yes |
| Linux x86_64 (CI) | `gtk.linux.x86_64` | pass | yes, under xvfb |
| Linux arm64 (container) | `gtk.linux.aarch64` | pass | yes, under xvfb |
| Windows x86_64 (CI) | `win32.win32.x86_64` | pass | yes |

### Windows

TVRenamer on Windows

### macOS

TVRenamer on macOS

### Linux

TVRenamer on Linux

The macOS shot shows the `TVRenamer` and `Help` application menu, so `CocoaUIEnhancer`'s reflection into `org.eclipse.swt.internal.cocoa` still works on current SWT. Linux and Windows show the `File`/`Help` menu bar, the branch of `setupMenuBar` that macOS never takes.

The screenshots also caught a bug that no test would have: on the 1024x768 Windows runner the window was placed partly off the left edge, clipping the `File` menu and the `Add files` button. `positionWindow` offset the window from the right and bottom edges with no lower bound, so any display narrower or shorter than the window pushed its top left corner off screen. Fixed by clamping to the monitor's client area, which also stops the window landing under the taskbar or dock.

Note for anyone running the tests in a root container: `MoveTest.testFileMoverCannotMove`, `testMoveRunnerCannotMove`, the two `ConflictTest` equivalents and `FileUtilsTest.testEnsureWritableDirectoryCantWrite` all chmod a file read-only and assert the write fails. Root ignores permission bits, so those five fail as root and pass as any normal user.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,30 +1,67 @@
...
+    return "org.eclipse.swt.gtk.linux.${swtArch}"
+}()
 
-// Only for MacOS
-applicationDefaultJvmArgs = ["-XstartOnFirstThread"]
+def isMacOs = System.getProperty('os.name').toLowerCase().contains('mac')
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
```
[the remark this answers is not in the document]
Comment (vipuldelwadia) [reply build.gradle]: Confirmed and fixed in eb428d6.

Reproduced in an `eclipse-temurin:25-jdk` container, where the build failed before compiling anything:

```
Cannot find a Java installation on your machine (Linux aarch64) matching:
{languageVersion=21, vendor=any vendor, implementation=vendor-specific, nativeImageCapable=false}.
Toolchain download repositories have not been configured.
```

So the README claim of "21 or later" was wrong.

Went with the first of your two suggestions, the current JDK plus a release target, rather than toolchain provisioning: it keeps the build working offline and avoids downloading a second JDK. The toolchain block is replaced by `options.release = 21`, plus a `taskGraph.whenReady` check so an older JDK gets a one-line message instead of a confusing compiler error.

Verified on JDK 21 locally and in a JDK 25 container, which now builds and passes all 53 tests.

File: src/main/java/org/tvrenamer/controller/TVmazeProvider.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    private static boolean apiIsDeprecated = false;
+
+    private static final String API_URL = "https://api.tvmaze.com/";
+
+    // The URL to get, to receive options for a given series search string.
+    private static final String BASE_SEARCH_URL = API_URL + "search/shows?q=";
+
+    // The URL to get, to receive the episode listing for a specific series.
+    private static final String BASE_LIST_URL = API_URL + "shows/";
+    private static final String BASE_LIST_SUFFIX = "/episodes";
```
[the remark this answers is not in the document]
Comment (vipuldelwadia) [reply src/main/java/org/tvrenamer/controller/TVmazeProvider.java:57]: The premise is right but the conclusion does not hold, so I have documented this instead of changing the request.

`specials=1` does return more episodes (Robot Chicken goes from 218 to 232). But it does not make `S00Exx` resolvable, because **TVmaze has no season 0**:

```
episodes with null number: 14
 season=1 number=None type=significant_special name=Christmas Special
 season=2 number=None type=significant_special name=Star Wars Episode I
distinct seasons: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```

TVmaze files a special under the season it aired in, with a null episode number. `Episode` maps a null number to a null `airPlacement`, and `Show.addEpisodeToSeason` skips any episode without a placement, so those 14 entries can never be matched to a file. Adding the parameter would download unusable entries and log an INFO line for each one, while `S00Exx` filenames would remain unresolvable.

You are right that this is a regression against the old TheTVDB feed, which did carry season 0. Matching specials against TVmaze needs a different key, title or air date rather than placement, which is a feature rather than part of this build fix. A comment in `TVmazeProvider` now records both the reason and the limitation.
